tests/js/server/aql/aql-memory-limit.js

```diff
@@ -95,16 +87,226 @@ function ahuacatlMemoryLimitTestSuite () {
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " SORT doc._key RETURN doc", null, { memoryLimit: 10 * 1000 * 1000 }).json;
+      assertEqual(100000, actual.length);
+        
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " SORT doc._key RETURN doc", null, { memoryLimit: 5 * 1000 * 1000 });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+
+    testSort : function () {
+      // turn off constrained heap sort
+      const optimizer = { rules: ["-sort-limit"] };
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " SORT doc.value1 LIMIT 10 RETURN doc", null, { memoryLimit: 15 * 1000 * 1000, optimizer }).json;
+      assertEqual(10, actual.length);
+        
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " SORT doc.value1 LIMIT 10 RETURN doc", null, { memoryLimit: 10 * 1000 * 1000, optimizer });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+    
+    testCollect1 : function () {
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " COLLECT v = doc.value1 OPTIONS { method: 'hash' } RETURN v", null, { memoryLimit: 10 * 1000 * 1000 }).json;
+      assertEqual(100000, actual.length);
+      
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " COLLECT v = doc.value1 OPTIONS { method: 'hash' } RETURN v", null, { memoryLimit: 5 * 1000 * 1000 });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+    
+    testCollect2 : function () {
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " COLLECT v = doc.value2 OPTIONS { method: 'hash' } RETURN v", null, { memoryLimit: 1 * 1000 * 1000 }).json;
+      assertEqual(10, actual.length);
+      
+      actual = AQL_EXECUTE("FOR doc IN " + cn + " COLLECT v = doc.value2 OPTIONS { method: 'hash' } RETURN v", null, { memoryLimit: 100 * 1000 }).json;
+      assertEqual(10, actual.length);
+        
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " COLLECT v = doc.value2 OPTIONS { method: 'hash' } RETURN v", null, { memoryLimit: 10 * 1000 });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+  };
+}
 
+function ahuacatlMemoryLimitGraphQueriesTestSuite () {
+  const vn = "UnitTestsVertex";
+  const en = "UnitTestsEdge";
+
+  return {
+    setUpAll : function () {
+      db._drop(en);
+      db._drop(vn);
+      
+      const n = 400;
+
+      // only one shard because that is more predictable for memory usage
+      let c = db._create(vn, { numberOfShards: 1 });
+
+      let docs = [];
+      for (let i = 0; i <= n; ++i) {
+        docs.push({ _key: "test" + i });
+      }
+      c.insert(docs);
+
+      c = db._createEdgeCollection(en, { numberOfShards: 1 });
+
+      const weight = 1;
+      
+      docs = [];
+      for (let i = 0; i < n; ++i) {
```

Comment:
Short note here will also help in the future to directly see which actual graph will be built up here.